### PR TITLE
Make ink_sparkle use FragmentShader

### LINKED ISSUE #104312: Incorrect InkSparkle colors
## Steps to Reproduce

1. Make an `InkWell`. Specify `Colors.blue` as `splashColor`.
2. Witness a green splash (incorrect).
3. Change the color to `Colors.green`.
4. Witness a blue splash (incorrect).
5. Change the color to `Colors.red`.
6. Witness a red splash (correct).

This is due to a bug in `ink_sparkle.dart` that has the blue and green components reversed in the `_colorToVector4` function. I am submitting a PR with a fix as well.

### PR DESCRIPTION
**Pull request by zanderso**
Framework side of https://github.com/flutter/engine/pull/35253

Fixes https://github.com/flutter/flutter/issues/104312

### CODE REVIEW

File: packages/flutter/lib/src/material/ink_sparkle.dart [hunk 1]
```diff
@@ -298,68 +312,59 @@ class InkSparkle extends InteractiveInkFeature {
...
-  }
-
-  Vector4 _colorToVector4(Color color) {
-    return Vector4(
-        color.red / 255.0,
-        color.blue / 255.0,
-        color.green / 255.0,
-        color.alpha / 255.0,
-      );
+    _fragmentShader
```
Comment (jonahwilliams) [inline packages/flutter/lib/src/material/ink_sparkle.dart:329]: This looks really nice

File: packages/flutter/lib/src/material/ink_sparkle.dart [hunk 2]
```diff
@@ -298,68 +312,59 @@ class InkSparkle extends InteractiveInkFeature {
...
+      ..floatUniforms[0] = _color.red / 255.0       // uColor
+      ..floatUniforms[1] = _color.blue / 255.0
+      ..floatUniforms[2] = _color.green / 255.0
+      ..floatUniforms[3] = _color.alpha / 255.0
```
Comment (jonahwilliams) [inline packages/flutter/lib/src/material/ink_sparkle.dart:333]: Shouldn't this be `rgba`? accidental transposition of green and blue maybe.

Comment (MrBirb) [reply packages/flutter/lib/src/material/ink_sparkle.dart:333]: [#104312](https://github.com/flutter/flutter/issues/104312) sounds relevant.

File: packages/flutter/lib/src/material/ink_sparkle.dart [hunk 3]
```diff
@@ -298,68 +311,70 @@ class InkSparkle extends InteractiveInkFeature {
...
-    return Vector4(
-        color.red / 255.0,
-        color.blue / 255.0,
-        color.green / 255.0,
-        color.alpha / 255.0,
-      );
+    _fragmentShader
+      // uColor
+      ..setFloat(0, _color.red / 255.0)
+      ..setFloat(1, _color.green / 255.0)
```
Comment (zanderso) [inline packages/flutter/lib/src/material/ink_sparkle.dart:331]: Fixed the order here.

**Review by jonahwilliams — APPROVED** [review]
LGTM!

File: packages/flutter/lib/src/material/ink_sparkle.dart (hunk 3 above)
Comment (clocksmith) [reply packages/flutter/lib/src/material/ink_sparkle.dart:331]: woops, nice!

Comment (ArZargaryan) [reply packages/flutter/lib/src/material/ink_sparkle.dart:331]: > Fixed the order here.

This is not fixed in 3.3.10(latest stable channel).
So when will this be fixed?

**Pull request merged**